### PR TITLE
Remove gap skipping when opening engine

### PR DESCRIPTION
**Pull request by jasontedor**
Today when opening the engine we skip gaps in the history, advancing the local checkpoint until it is equal to the maximum sequence number contained in the commit. This allows history to advance, but it leaves gaps. A previous change filled these gaps when recovering from store, but since we were skipping the gaps while opening the engine, this change had no effect. This commit removes the gap skipping when opening the engine allowing the gap filling to do its job.

Relates #10708, relates #24238

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java [hunk 1]
```diff
@@ -226,21 +217,17 @@ public InternalEngine(EngineConfig engineConfig) throws EngineException {
...
+    public int fillSeqNoGaps(long primaryTerm) throws IOException {
+        try (ReleasableLock ignored = writeLock.acquire()) {
             ensureOpen();
-            final long localCheckpoint = seqNoService.getLocalCheckpoint();
-            final long maxSeqId = seqNoService.getMaxSeqNo();
+            final long localCheckpoint = seqNoService().getLocalCheckpoint();
+            final long maxSeqNo = seqNoService().getMaxSeqNo();
             int numNoOpsAdded = 0;
-            for (long seqNo = localCheckpoint + 1; seqNo <= maxSeqId;
-                 // the local checkpoint might have been advanced so we are leap-frogging
```
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java]: why lose the comment?

**Review by bleskes — APPROVED** [review]
LGTM

File: core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java (hunk 1 above)
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java]: Back in 61a209339315dfdd3d40d03f0e2127107d7911c1.

**Comment by jasontedor** [comment]
Thanks @bleskes.

**Pull request merged**